### ISSUE TITLE
[StructuralMechanicsApplication] Error: can not compute the center of a geometry of zero points

### ISSUE THREAD

**Issue by Joekchu** [KratosMultiphysics/Kratos#8825]
Dear all,

I am using the structural mechanics application with my own customized utility. In order to initialize my utility, a customized static solver is also implemented. However, when I run a simple case, I met an error with a message like "Error: can not compute the center of a geometry of zero points". The error and solver are attached, please kindly have a look for your deeper understanding. 

Does anyone ever experience this problem? Your reply will be invaluable to me! Thank you very much!

Best regards,
Joe

[Customised_solver.txt](https://github.com/KratosMultiphysics/Kratos/files/6561623/Customised_solver.txt)
[Error_Messages.txt](https://github.com/KratosMultiphysics/Kratos/files/6561624/Error_Messages.txt)

**Comment by loumalouomega** [comment]
Questions after your solver, are you redefining the variables in your application?

**Comment by loumalouomega** [comment]
Looks like you are calling a method of a base class, is that geometry properly constructed?

**Comment by Joekchu** [comment]
@loumalouomega Yes, Vicente, according to documents and tutorial, I redefine the variables and also link the utility and application by pybind11. 

Could you please kindly elaborate more on "is that geometry properly constructed?"

Thank you very much!

**Comment by loumalouomega** [comment]
> @loumalouomega Yes, Vicente, according to documents and tutorial, I redefine the variables and also link the utility and application by pybind11.
> 
> Could you please kindly elaborate more on "is that geometry properly constructed?"
> 
> Thank you very much!

Which tutorial?, I think it is not necessary to redefine the varibles already existing in other apps and the core...

**Comment by loumalouomega** [comment]
> > Could you please kindly elaborate more on "is that geometry properly constructed?"

Looks like the element/condition you are using is using the default geometry, and therefore there are no nodes to even compute anything

**Comment by Joekchu** [comment]
> > @loumalouomega Yes, Vicente, according to documents and tutorial, I redefine the variables and also link the utility and application by pybind11.
> > Could you please kindly elaborate more on "is that geometry properly constructed?"
> > Thank you very much!
> 
> Which tutorial?, I think it is not necessary to redefine the varibles already existing in other apps and the core...

Cuz I don't know whether my variables were already defined in the core..so I may repeatedly define some variables. Is it affecting the running?



> > > Could you please kindly elaborate more on "is that geometry properly constructed?"
> 
> Looks like the element/condition you are using is using the default geometry, and therefore there are no nodes to even compute anything

I check the condition and find it seems that I am using the default geometry in the construction. Please find the attached construction of my condition codes.
[Constructed function.txt](https://github.com/KratosMultiphysics/Kratos/files/6561864/Constructed.function.txt)
